### ISSUE TITLE
Component.onGameResize with zero value first call

### ISSUE THREAD

**Issue by rusoleal** [flame-engine/flame#658]
## Description

first onGameResize on all components added to game comes with size(0.0,0.0). As consecuence, the onscreen joystick component isn't draw until next onGameResize call.

I/flutter (28817): player resize: [0.0,0.0]
I/flutter (28817): preload
I/flutter (28817): joystick resize: [0.0,0.0]
I/flutter (28817): resizing: [1280.0,844.0]
I/flutter (28817): load
I/flutter (28817): postload
I/flutter (28817): resizing: [1280.0,844.0]

## Development environment

Flame version:
  flame: ^1.0.0-rc6

Flutter doctor:

[✓] Flutter (Channel stable, 1.22.6, on macOS 11.2 20D64 darwin-x64, locale en-ES)
    • Flutter version 1.22.6 at /Users/rubenleal/Projects/flutter
    • Framework revision 9b2d32b605 (3 weeks ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/rubenleal/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/rubenleal/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[!] VS Code (version 1.53.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • sdk gphone x86 arm (mobile)    • emulator-5554                        • android-x86 • Android 11 (API 30) (emulator)
    • iPad (8th generation) (mobile) • 567333F8-915B-46F6-9B11-D1BABA5DC02C • ios         • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    ! Error: iPad de Ruben is not connected. Xcode will continue when iPad de Ruben is connected. (code -13)

! Doctor found issues in 2 categories.
## Runtime

This issue is related to running on which platform? (Select all that apply)

 - [ x] Android :android:
 - [ x] iOS :apple:
 - [ ] Web :browser:

## Minimal reproducible code (Required for bugs)

joystick demo at https://github.com/flame-engine/flame/tree/master/doc/examples/joystick

**Comment by erickzanardo** [comment]
@rusoleal that example is incorrect, we need to fix it. Components shouldn't be added on the game constructor but on the `onLoad` function.

We will add validations to prevent this confusion in the future.

**Comment by rusoleal** [comment]
Now is perfect!!! Thanks Erick. 

Im working actively with flame and I would like to contribute as possible.

I try by myself to fix the problem and I created my custom game deriving directly from Game. My problem was that using JoystickController depends on BaseGame because...

abstract class JoystickController extends Component with HasGameRef<BaseGame> {

I checked that JoystickController just need the canvas size, then can have a HasGameRef<Game> not limiting the use to derived BaseGame apps.

Have a nice day.
